yesod-auth/Yesod/Auth/Email.hs

```diff
@@ -141,11 +144,18 @@ registerR = PluginR "email" ["register"]
 forgotPasswordR = PluginR "email" ["forgot-password"]
 setpassR = PluginR "email" ["set-password"]
 
+verifyURLHasSetPassText :: Text
+verifyURLHasSetPassText = "has-set-pass"
+
 -- |
 --
 -- @since 1.4.5
-verifyR :: Text -> Text -> AuthRoute -- FIXME
-verifyR eid verkey = PluginR "email" ["verify", eid, verkey]
+verifyR :: Text -> Text -> Bool -> AuthRoute -- FIXME
+verifyR eid verkey hasSetPass = PluginR "email" [ "verify"
+                                                , eid
+                                                , verkey
+                                                , if hasSetPass then verifyURLHasSetPassText else ""
```

Comment:
This seems like it will unnecessarily change the URLs with a trailing slash, which may break things at the path cleanup level. How about something like `"verify":eid:verkey:(if hasSetPass then [verifyURLHasSetPassText] else [])`?